Now, as I really WANT a single-iteration loop, I could put the previous construct back but I'd be fighting against the shellcheck bot:
https://www.shellcheck.net/wiki/SC2043

--- a/scripts/prepare_commit.sh
+++ b/scripts/prepare_commit.sh
@@ -57,7 +57,22 @@ if [ -z "$MODIFIED" ]; then
   exit 0
 fi
 
-if [[ -n "$QGIS_CHECK_SPELLING" && -x "${TOPLEVEL}"/scripts/spell_check/check_spelling.sh ]]; then "${TOPLEVEL}"/scripts/spell_check/check_spelling.sh "$MODIFIED"; fi
+# Run spell checker if requirements are met
+if ! command -v ag > /dev/null; then
+  echo "WARNING: the ag(1) executable was not found, spell checker could not run" >&2
+elif ! command -v unbuffer > /dev/null; then
+  echo "WARNING: the unbuffer(1) executable was not found, spell checker could not run" >&2
+else
+  "${TOPLEVEL}"/scripts/spell_check/check_spelling.sh "$MODIFIED"
+fi
+
+# Run shell checker if requirements are met
+if command -v shellcheck > /dev/null; then
+  # TODO: allow passing $MODIFIED
+  ${TOPLEVEL}/tests/code_layout/test_shellcheck.sh || exit 1
+else
+  echo "WARNING: the shellcheck(1) executable was not found, shell checker could not run" >&2
+fi
 
 FILES_CHANGED=0
 

